src/App.tsx

```diff
@@ -1,61 +1,143 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { useState } from 'react';
+
+interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user?: User;
+}
+
+function getUserById(userId: number): User | undefined {
+  return usersFromServer.find(user => user.id === userId);
+}
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(
+    todosFromServer.map(todo => ({
+      ...todo,
+      user: getUserById(todo.userId),
+    })),
+  );
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(true);
+  const [userId, setUserId] = useState(0);
+  const [hasUserError, setHasUserError] = useState(true);
```

Comment:
Similar to `hasTitleError`, the initial state for `hasUserError` is set to `true`. It might be better to initialize it to `false` and update it based on the validation logic.